src/services/CertificateGenerationService.ts

```diff
@@ -114,35 +131,55 @@ class CertificateGenerationService {
 	}
 
 	/**
-	 * Handler method for retrieving feature flags and checking if test station is in Wales
+	 * Retrieve feature flags by using or setting cache
+	 */
+	public async retrieveAndSetFlags() {
+		if (CertificateGenerationService.flags) {
+			console.log('Feature flag cache already set');
+			return;
+		} else {
+			try {
+				console.log('Feature flag cache not set, retrieving feature flags');
+				CertificateGenerationService.flags = await getProfile();
+			} catch (e) {
+				console.error(`Failed to retrieve feature flags - ${e}`);
+			}
+		}
+		console.log('Using feature flags ', CertificateGenerationService.flags);
+	}
+
+	/**
+	 * Determine if Welsh translation is required for the given test result
 	 * @param testResult
+	 * @param featureFlags
```

Comment:
is this needing removed? isn't a param anymore

Reply:
Done and repushed.